src/workflows-app/SubmissionConfig.test.js

```diff
@@ -164,6 +164,33 @@ describe('BaseSubmissionConfig renders workflow details', () => {
     expect(screen.getByText('Workflow Script')).toBeInTheDocument();
   });
 
+  it('should render a functional call cache toggle button', async () => {
+    await act(async () => {
+      return render(
+        h(BaseSubmissionConfig, {
+          methodId: '123',
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+    const user = userEvent.setup();
+    const callCacheToggleButton = screen.getByTestId('call-cache-toggle'); // The switch itself
```

Comment:
You only want to define a test ID and use it to get a component in a test when there is no other way to get the component… in general, you should get the component the same way a user would (which enforces accessibility).

Since you now have a label linked to the switch, just use `screen.getByLabelText('Call Caching:')` to get the switch and interact directly with it that way. To be clear, what is being returned by that method is NOT the label, it's the interactive input identified by the label. Then you can delete the test ID from the actual code.